comp_sci_gender_bias/pipeline/glove_differences/make_mean_differences.py

```diff
@@ -31,33 +34,81 @@ def calc_mean_gender_diff(
             with associated POS and Corpus label
         glove_dists: GloveDistances class object
         data_source_lbl: Data source label
-        subject_lbl: Subject label
+        subject: Subject
+        word_removal: Subject related words to remove.
+            "crucial", "optional" or None
 
     Returns:
         Dataframe with columns for:
             - POS
             - mean_gender_diff
             - subject
             - data_source
+            - words_removed
     """
-    glove_sims = glove_dists.gender_similarity_difference_word_list(
-        sub_word_pos_corpus["Word"].tolist()
-    )
+    if word_removal is not None and subject != "drama":
+        words_to_remove = subject_specific_words(
+            subject=subject, specific_word_type=word_removal
+        )
+    else:
+        words_to_remove = []
+
+    words = [
+        word
+        for word in sub_word_pos_corpus["Word"].tolist()
+        if word not in words_to_remove
+    ]
+    glove_sims = glove_dists.gender_similarity_difference_word_list(words)
     sub_word_pos_corpus["Male - Female"] = sub_word_pos_corpus["Word"].map(glove_sims)
     sub_word_pos_corpus = sub_word_pos_corpus.dropna()
+    row_length = len(POS_LABELS)
     return pd.DataFrame.from_dict(
         {
             "POS": POS_LABELS,
             "mean_gender_diff": [
                 sub_word_pos_corpus.query(query)["Male - Female"].mean()
                 for query in POS_QUERIES
             ],
-            "subject": [subject_lbl] * len(POS_LABELS),
-            "data_source": [data_source_lbl] * len(POS_LABELS),
+            "subject": [subject] * row_length,
+            "data_source": [data_source_lbl] * row_length,
+            "words_removed": [word_removal] * row_length,
```

Comment:
pandas will expand scalar values to fill the length of the dataframe on construction
```suggestion
            "subject": subject,
            "data_source": data_source_lbl,
            "words_removed": word_removal,
```

Reply:
Ahh, I didn't realise, great!